packages/aws-rfdk/lib/lambdas/nodejs/lib/configure-spot-event-plugin/test/spot-event-plugin-client.test.ts

```diff
@@ -238,4 +256,94 @@ describe('SpotEventPluginClient', () => {
     // THEN
     await expect(promise).rejects.toEqual(statusMessage);
   });
+
+  test.each([
+    [CollectionType.Group, successfulGroupResponse, groupsColection],
+    [CollectionType.Pool, successfulPoolResponse, poolsColection],
+  ])('Successful getCollection for %s', async (type: CollectionType, response: Response, expectedResult: any) => {
+    // WHEN
+    // eslint-disable-next-line dot-notation
+    spotEventPluginClient['deadlineClient'].GetRequest = jest.fn().mockResolvedValue(response);
+    const result = await spotEventPluginClient.getCollection(type);
+
+    // THEN
+    expect(result).toEqual(expectedResult);
+    // eslint-disable-next-line dot-notation
+    expect(spotEventPluginClient['deadlineClient'].GetRequest).toBeCalledTimes(1);
+    expect(consoleLogMock.mock.calls.length).toBe(1);
+    expect(consoleLogMock.mock.calls[0][0]).toMatch(`Getting ${type} collection:`);
+  });
+
+  test('failed getCollection', async () => {
+    // GIVEN
+    const statusMessage = 'error message';
+
+    // eslint-disable-next-line dot-notation
+    spotEventPluginClient['deadlineClient'].GetRequest = jest.fn().mockRejectedValue(new Error(statusMessage));
+    const promise = spotEventPluginClient.getCollection(CollectionType.Group);
+
+    // THEN
+    await expect(promise).rejects.toThrowError(`Failed to get group collection. Reason: ${statusMessage}`);
+  });
+
+  test('failed getCollection with invalid response', async () => {
+    // GIVEN
+    const invalidGroupResponse = {
+      data: {
+        Pools: {},
+      },
+    };
+
+    // WHEN
+    // eslint-disable-next-line dot-notation
+    spotEventPluginClient['deadlineClient'].GetRequest = jest.fn().mockResolvedValue(invalidGroupResponse);
+    const promise = spotEventPluginClient.getCollection(CollectionType.Group);
+
+    // THEN
+    await expect(promise).rejects.toThrowError(`Failed to get group collection. Reason: Failed to receive a group collection. Invalid response: ${JSON.stringify(invalidGroupResponse.data)}.`);
+  });
+
+  test('successful saveCollection for groups', async () => {
+    // WHEN
+    // eslint-disable-next-line dot-notation
+    spotEventPluginClient['deadlineClient'].PostRequest = jest.fn().mockResolvedValue({});;
+    const result = await spotEventPluginClient.saveCollection( groupsColection, CollectionType.Group);
+
+    // THEN
+    expect(result).toBeTruthy();
+    // eslint-disable-next-line dot-notation
+    expect(spotEventPluginClient['deadlineClient'].PostRequest).toBeCalledTimes(1);
+    expect(consoleLogMock.mock.calls.length).toBe(2);
+    expect(consoleLogMock.mock.calls[0][0]).toMatch(/Saving group collection:/);
+    expect(consoleLogMock.mock.calls[1][0]).toBe(groupsColection);
+  });
+
+  test('successful saveCollection for pools', async () => {
```

Comment:
We can do the same thing we did for `Successful getCollection for %s` here, using `test.each` to combine this test and the groups version.